engine/transaction/draft/create_draft.go

```diff
@@ -0,0 +1,35 @@
+package draft
+
+import (
+	"context"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+)
+
+// Create creates a new draft transaction based on specification.
+func Create(ctx context.Context, spec *TransactionSpec) (*Transaction, error) {
+	tx := &sdk.Transaction{}
+	if spec == nil {
+		return nil, txerrors.ErrDraftSpecificationRequired
+	}
+	outputs, annotations, err := spec.outputs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tx.Outputs = outputs
```

Comment:
The `tx` is unnecessarily created at the top. 
It could be done here (where tx.Outputs line currently is):

```suggestion
	tx := &sdk.Transaction{
	    Outputs: outputs,
	}
```